File: [134]加油站.cpp
#include <vector>
using namespace std;

class Solution {
public:
	int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
		int idx = -1;

		vector<int> values = gas;
		for (int i = 0; i < values.size(); ++i) {
			values[i] -= cost[i];
		}

		int cum = 0;
		for (int i = 0; i < values.size(); ++i) {
			if (cum <= 0) {
				if (values[i] >= 0) {
					idx = i;
					cum = values[i];
				}
				else
					continue;
			}
			else
				cum += values[i];
		}

		if (cum < 0 || idx == -1)
			return -1;
		else {
			int cum2 = 0;
			for (int i = 0; i < idx; ++i)
				cum2 += values[i];
			if (cum2 + cum >= 0)
				return idx;
			else
				return -1;
		}
	}
};

File: solutions/[134]加油站.py
from typing import List


class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        i = 0

        while i < len(gas):
            sum = 0
            for pj in range(i, i+len(gas)):
                j = pj % len(gas)
                sum += gas[j] - cost[j]
                if sum < 0:
                    i = pj + 1
                    break
                if pj - i + 1 == len(gas):
                    return i
                print(i, j)
        
        return -1